kong/api/routes/kong.lua

```diff
@@ -158,6 +158,23 @@ return {
       return kong.response.exit(200, status_response)
     end
   },
+  ["/endpoints"] = {
+    GET = function(self, dao, helpers)
+      local endpoints = setmetatable({}, cjson.array_mt)
+      local lapis_endpoints = require("kong.api").ordered_routes
+
+      for k, v in pairs(lapis_endpoints) do
+        if type(k) == "string" then -- skip numeric indices
+          endpoints[#endpoints + 1] = k:gsub(":([^/:]+)", function(m)
+              return "{" .. m .. "}"
+            end)
+        end
+      end
+      table.sort(endpoints)
```

Comment:
I'd prefer a segment aware sorting?

Not this:
```
/a
/ab
/ab/a
/a/z
```

But this:
```
/a
/a/z
/ab
/ab/a
```

Reply:
This should do the trick:

```
table.sort(t, function(a, b)
  return  a:gsub("/", "\x00") < b:gsub("/", "\x00")
end)
```